docs/sdk-configuration.md

```diff
@@ -0,0 +1,492 @@
+# Design for configuring the SDK
+
+This document outlines some of the goals we have for user configuration of the SDK. It is a
+continuation of discussion started with https://github.com/open-telemetry/opentelemetry-java/issues/2022.
+
+## Target audiences
+
+There are a few different target audiences that are related to our configuration story.
+
+- Application developers, aka end-users. Often have no knowledge of tracing but want to add 
+OpenTelemetry to their app and see traces show up in a console. Application developers will increase
+in number forever, while the below are more constant.
+
+- Dev-ops / framework developers. Write components or frameworks to support providing tracing to
+their application developers. May write custom SDK extensions such as exporters, samplers, to fit
+with their internal infrastructure and as such have some familiarity with tracing at least as the
+SDK presents it.
+
+- OpenTelemetry maintainers. Write the SDK code.
+
+When making decisions, especially about complexity, we always prioritize the application developers,
+then framework developers, then maintainers. This is because we expect those with less domain
+knowledge about tracing to require a simpler experience than those with more. There is also more
+bang-for-the-buck by making the end-user experience as streamlined as possible since we expect there
+to be much more of them than other audiences.
+
+## Goals and non-goals
+
+### Goals
+
+- Provide a single entrypoint to configuring the SDK. For end-users, less familiar with the SDK, we
+want to have everything together to provide discoverability and simpler end-user code.
```

Comment:
is this a "single entrypoint" per use-case, or only a single entrypoint, period?